### DISCUSSION TITLE
Delete file by name

### DISCUSSION THREAD

**Discussion by VC-dell** [iterative/PyDrive2#357] [Q&A] [answered]
How do I delete a file on google drive, knowing only its name but not the file id?

**Comment by shcheklein** [comment, marked as answer]
Hi @VC-dell ! It's possible to do the listing first to get IDs (e.g. here https://docs.iterative.ai/PyDrive2/filemanagement/#get-files-by-complex-queries ). And then use that ID to delete the file. Also, keep in mind - Google allows duplicate names - so you might receive multiple IDs back for a single query.

**Reply by VC-dell** [reply]
I have thought of that solution, but will it be too intensive, assuming that I have about 20000 files?

**Reply by shcheklein** [reply]
I think you can pass multiple titles (names) at once or if this is a directory use `parent` condition to list files inside
